contracts/delegatecall/AssetTransferToENS.sol

```diff
@@ -0,0 +1,86 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.20;
+
+import {ENS} from "@ensdomains/ens-contracts/contracts/registry/ENS.sol";
+import {AddrResolver} from "@ensdomains/ens-contracts/contracts/resolvers/profiles/AddrResolver.sol";
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+import {LibAddress} from "../library/LibAddress.sol";
+
+contract AssetTransferToENS {
+    using LibAddress for address;
+    using SafeERC20 for IERC20;
+
+    ENS immutable _ens;
+
+    constructor(ENS ens) {
+        _ens = ens;
+    }
+
+    function sendCallToENS(
```

Comment:
Maybe `sendMessageToENS` or `callENS`?

Reply:
since it can also be a simple ether transfer, `sendMessageToENS` probably doesn't fit. 
Between `sendCallToENS` and `callENS`, I think I prefer `sendCallToENS ` because all other functions in the contract have names like "sendxxToENS"

Reply:
I don't know. "Send call" just sounds weird to me, because you wouldn't normally say this in English. You either send something to someone, like a message, or you call them.

> since it can also be a simple ether transfer, sendMessageToENS probably doesn't fit.

I don't understand. Aren't Ether transfers simply message calls with a non-zero amount of Wei?

> because all other functions in the contract have names like "sendxxToENS"

That's one of the reasons why I wanted this function to be in another contract, because it's too general. It's not only suited for Ether transfers, but for arbitrary message calls. If we only care about Ether transfers, we could simply remove the `payload` argument and rename it as `sendEtherToENS`.

Reply:
`sendEtherToENS` should cover most of the cases right? That was the main purpose.
And sending arbitrary calls was just a by-product.

Reply:
Ok, let's rename it to `sendEtherToENS`. But should we keep the `payload` parameter? It would only be used to call a function of an ENS-identifiable contract, which is very specific, and unlikely, I presume.

Reply:
Yes. Let's keep it. Users can decide what to do with it :)